frontend/src/hooks/fetchUser.ts

```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from 'react'
+import axios from 'axios'
+
+export const fetchUser = (): any => {
+  const [user, setUser] = useState<any>(null)
+  const [loadingStatus, setLoadingStatus] = useState<string>('unloaded')
+
+  useEffect(() => {
+    const getUserData = async (): Promise<void> => {
+      try {
+        setLoadingStatus('loading')
+        const userDataResponse = await axios.get('http://localhost:8000/userData/1', { withCredentials: true })
+        setUser(userDataResponse.data)
+        setLoadingStatus('loaded')
+      } catch (error) {
+        setLoadingStatus('error')
+        console.error(error)
+      }
+    }
+    void getUserData()
```

Comment:
does this need to even get wrapped in a fn? what benefit do we get in defining the fn and then calling it here rather than calling this code directly or pulling it out into a separate fn outside this component?